code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,322 @@
+
+/// Subtype that's always junkmail
+/obj/item/mail/junkmail/Initialize()
+	. = ..()
+	junk_mail()
+
+/obj/structure/closet/crate/mail
+	name = "mail crate"
+	desc = "A certified post crate from CentCom."
+	icon_state = "mail_crate"
+	door_anim_time = 0
+
+/* Fills this mail crate with N pieces of mail, where N is the lower of the amount var passed,
+** and the maximum capacity of this crate. If N is larger than the number of alive human players, the excess will be junkmail.*/
+/obj/structure/closet/crate/mail/proc/populate(amount)
+	var/mail_count = min(amount, storage_capacity)
+	var/list/mail_recipients = list() //fills the crate for the recipients
+
+	for(var/mob/living/carbon/human/human in GLOB.player_list)
+		// Skip wizards, nuke ops, cyborgs and dead people; Centcom does not send them mail
+		if(human.stat == DEAD || !human.mind || human.mind.assigned_role == "Cyborg" || human.mind.special_role)
+			continue
+
+		mail_recipients += human.mind
+
+	if(mail_count < 15)
+		for(var/i in 1 to rand(3,8))
+			var/obj/item/mail/new_mail
+			if(prob(FULL_CRATE_LETTER_ODDS))
+				new_mail = new /obj/item/mail(src)
+			else
+				new_mail = new /obj/item/mail/envelope(src)
+			new_mail.junk_mail()
+
+	for(var/i in 1 to mail_count)
+		var/datum/mind/recipient = pick_n_take(mail_recipients)
+		var/obj/item/mail/new_mail
+		if(prob(FULL_CRATE_LETTER_ODDS))
+			new_mail = new /obj/item/mail(src)
+		else
+			new_mail = new /obj/item/mail/envelope(src)
+		if(recipient)
+			new_mail.initialize_for_recipient(recipient)
+		else
+			new_mail.junk_mail()
+
+	update_icon()
+
+/// Crate for mail that automatically depletes the economy subsystem's pending mail counter.
+/obj/structure/closet/crate/mail/economy/Initialize()
+	. = ..()
+	populate(SSeconomy.mail_waiting)
+	SSeconomy.mail_waiting = 0
+
+/// Crate for mail that automatically generates a lot of mail. Usually only normal mail, but on lowpop it may end up just being junk.
+/obj/structure/closet/crate/mail/full
+	name = "brimming mail crate"
+	desc = "A certified post crate from CentCom. Looks stuffed to the gills."
+
+/obj/structure/closet/crate/mail/full/Initialize()
+	. = ..()
+	populate(INFINITY)
+
+/obj/item/paper/fluff/junkmail_redpill
+	name = "smudged paper"
+	icon_state = "scrap"
+	var/nuclear_option_odds = 0.1
+
+/obj/item/paper/fluff/nice_argument
+	name = "RE: Nice Argument..."
+	icon_state = "paper"
+
+/obj/item/paper/fluff/nice_argument/Initialize()
+	. = ..()
+	var/station_name = station_name()
+	info ="Nice argument, however there's a <i>small detail</i>...<br>IP: '[rand(0,10)].[rand(0,255)].[rand(0,255)].[rand(0,255)]'<br> Station name: '[station_name]'<br>"
+
+/obj/item/paper/fluff/junkmail_redpill/Initialize()
+	. = ..()
+	if(!prob(nuclear_option_odds)) // 1 in 1000 chance of getting 2 random nuke code characters.
+		info = "<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[rand(0,9)][rand(0,9)][rand(0,9)]...'"
+		return
```

Comment:
nuke code has 6 digits iirc, it would be funny if you gave out like 4 of them
also it makes it obvious that the codes are fake if fake ones give you 3 numbers and real give you 2